Core/include/Acts/Propagator/DirectNavigator.hpp

```diff
@@ -94,17 +93,21 @@ class DirectNavigator {
   struct State {
     /// Externally provided surfaces - expected to be ordered
     /// along the path
-    SurfaceSequence surfaceSequence = {};
+    SurfaceSequence navSurfaces = {};
 
     /// Iterator the the next surface
-    SurfaceIter nextSurfaceIter = surfaceSequence.begin();
+    SurfaceIter navSurfaceIter = navSurfaces.begin();
 
     /// Navigation state - external interface: the start surface
     const Surface* startSurface = nullptr;
     /// Navigation state - external interface: the current surface
     const Surface* currentSurface = nullptr;
     /// Navigation state - external interface: the target surface
     const Surface* targetSurface = nullptr;
+    /// Navigation state - starting layer
+    const Layer* startLayer = nullptr;
```

Comment:
Does the `startLayer` and `targetLayer` makes sense for the direct navigator?

Reply:
Not necessarily, but the KalmanFitter needs the navigator to have these objects as it checks whether or not they are null. I think it is above my expertise level to know how to change the KF to not check these things, and for what purpose it serves. Anyway, my understanding is that this will go away once the layered geometry is removed also.

Reply:
I see. It makes sense to have there then.